### PR TITLE
docs(contributing): improve base & add component guidelines

### PR DESCRIPTION
**Pull request by zettca**
- Improve `CONTRIBUTING.md` and use it to populate `Contributing.stories.mdx`
- Add Component Guidelines doc

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -14,70 +14,44 @@ We welcome contributions to our project. Here are a few guidelines that will hel
```
Comment (MEsteves22) [inline CONTRIBUTING.md:14]: A few of the links on the page are not working, namely the ones for the contribution page and the links on the table of content.

File: docs/overview/community/ComponentGuidelines.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+## File Structure & Purpose
+
+The directory should contain the following files:
+
+```sh
+MyComp/
+├── index.ts            # entry file with the component's public exports
+├── MyComp.tsx          # component implementation and types
+├── MyComp.stories.tsx  # component documentation and examples
+├── MyComp.styles.ts    # component classes and stying utils
```
Comment (MEsteves22) [inline docs/overview/community/ComponentGuidelines.stories.mdx:19]: styling

File: docs/overview/community/ComponentGuidelines.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    "& $button": {
+      cursor: "not-allowed",
+    },
+  },
+  /** Applied to the button element */
+  button: {},
+});
+```
+
+```tsx
```
Comment (MEsteves22) [inline docs/overview/community/ComponentGuidelines.stories.mdx:79]: Do you think we should add the export for the MyComp classes type and static classes: `MyCompClasses` and `myCompClasses`? We do this in every component.

File: docs/overview/community/ComponentGuidelines.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+```sh
+MyComp/
+├── index.ts            # entry file with the component's public exports
+├── MyComp.tsx          # component implementation and types
+├── MyComp.stories.tsx  # component documentation and examples
+├── MyComp.styles.ts    # component classes and stying utils
+└── MyComp.test.tsx     # component unit tests
+```
+
+## Guidelines
```
Comment (MEsteves22) [inline docs/overview/community/ComponentGuidelines.stories.mdx:23]: Maybe we should have a bullet point saying that we also need to export the component on the `index.ts` file of the components folder?

File: docs/overview/community/ComponentGuidelines.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+- Components and their types MUST be named in PascalCase and prefixed with `Hv`.
+- Components MUST be exported from the `index.ts` file.
```
Comment (MEsteves22) [inline docs/overview/community/ComponentGuidelines.stories.mdx:26]: "Components and their types" MUST be exported from the `index.ts` file. => Just to make sure they don't forget the `MyCompProps` and `MyCompPropsClasses`.

**Review by MEsteves22 — COMMENTED** [review]
I really liked the guidelines 👌🏻🙏

I just have the following suggestions:

File: docs/overview/community/ComponentGuidelines.stories.mdx (hunk 3 above)
Comment (zettca) [reply docs/overview/community/ComponentGuidelines.stories.mdx:79]: yeah, makes sense 👍

File: docs/overview/community/ComponentGuidelines.stories.mdx (hunk 4 above)
Comment (zettca) [reply docs/overview/community/ComponentGuidelines.stories.mdx:23]: Isn't that the
> Components MUST be exported from the `index.ts` file.

I can rephrase it

**Review by MEsteves22 — APPROVED** [review]
LGTM 🙏

**Pull request merged**